Added resolve endpoint to secrets controller (#3538)
zendesk/samson: docs/secrets.md

### Deprecated

Secrets can be `deprecated` first and deleted when all use outside of samson (vault backend or accessing secrets via api client) has stopped.

### Resolving secrets from external tools

Secrets can be resolved in the context of a projects deploy group with the `/secrets/resolve.json` endpoint.

In the below example we have a secret configured for `a` in the deploy group `group1`, but no secret configured for `b`.

```
# params.json
{
  "project_id": 2,
  "deploy_group": "group1",
  "keys": [
    "a",
    "b"
  ]
}

> $ curl -s -d @params.json -H "Content-Type: application/json" -H "Authorization: Bearer $SAMSON_ACCESS_TOKEN" $SAMSON_BASE_URL/secrets/resolve.json | jq '.'

Author: yep, if you use keys like `keys[]=foo&keys[]=bar` ... in the query string

### Deprecated

Secrets can be `deprecated` first and deleted when all use outside of samson (vault backend or accessing secrets via api client) has stopped.

> ### Resolving secrets from external tools

Reviewer: ```suggestion
### Resolving secret keys to full paths via API
```

Secrets can be resolved to their full path in the context of a projects deploy group with the `/secrets/resolve.json` endpoint.

> In the below example we have a secret configured for `a` in the deploy group `group1`, but no secret configured for `b`.

Reviewer: ```suggestion
In the below example we have a secret configured for `keya` in the deploy group `group1`, but no secret configured for `keyb`.
```

> The endpoint supports both GET:

Reviewer: ```suggestion
The endpoint supports both GET and POST:
```

```
$ curl -s -H "Content-Type: application/json" -H "Authorization: Bearer $SAMSON_ACCESS_TOKEN" $SAMSON_BASE_URL/secrets/resolve.json?project_id=2&deploy_group=group1&keys[]=a&keys[]=b
{
  "a": "global/example-kubernetes/global/a",
  "b": null
}
```

> For ease of use to clients parameters can be sent in JSON format in a POST request. This way clients doesn't need to deal with url encoding semantics.

Reviewer: can just remove that line, the example is pretty obvious :)

```
# params.json
{
  "project_id": 2,
  "deploy_group": "group1",
  "keys": [
    "a",
    "b"
  ]
}

$ curl -s -d @params.json -H "Content-Type: application/json" -H "Authorization: Bearer $SAMSON_ACCESS_TOKEN" $SAMSON_BASE_URL/secrets/resolve.json | jq '.'
{
  "a": "global/example-kubernetes/global/a",
  "b": null
}
```

> This endpoint was added and is useful for resolving secrets as defined in kubernetes manifests, in the format that the `samson_secret_puller` or similar solutions use. The secrets are usually a list of names, with actual paths in the backend being different because of scoping/sharing concerns.

Reviewer: can remove that line too ... whoever reads finds the heading should know this already 🤷‍♂
